assets/in

```diff
@@ -134,26 +136,28 @@ if [ "$submodules" != "none" ]; then
     git submodule init $submodules
   fi
 
-  for submodule in $submodules; do
-    # remember submodule update config
-    update_conf_was_set=1
-    if update_conf="$(git config --get "submodule.${submodule}.update")"; then
-	    update_conf_was_set=0
-    fi
-
-    # temporarily set to our shallow clone deepening shell script
-    git config "submodule.${submodule}.update" "!$bin_dir"/deepen_shallow_clone_until_ref_is_found
-
-    concourse_git_resource__depth="$depth" \
-      git submodule update --no-fetch $depthflag $submodule_parameters $submodule
-
-    # restore submodule update config
-    if [ "$update_conf_was_set" != 0 ]; then
-      git config "submodule.${submodule}.update" "$update_conf"
-    else
-      git config --unset "submodule.${submodule}.update"
-    fi
-  done
+  if [ "$depth" -gt 0 ]; then
+    for submodule in $submodules; do
+      # remember submodule update config
+      update_conf_was_set=0
```

Comment:
Since `""` is an invalid value for `.update`, you might just use `[ -z $update_conf ]` below when deciding whether to set it, instead of this.

Reply:
Yes, you're right that that would work and it would be a bit simpler. Actually i considered that approach when i wrote the code originally, and thought that later Git may allow `""` as a valid value for `.update` and then that would break the code. So i thought it's better this way.

Reply:
If you ask me I think that's extremely unlikely - I doubt they'll add a recognition of the difference between "set to `""`" and "unset".  But I see you're going for future-proofing ... I have no opinion, then.  :)